partner_industry_secondary/models/res_partner_industry.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2015 Antiun Ingenieria S.L. - Javier Iniesta
+# Copyright 2016 Tecnativa S.L. - Vicent Cubells
+# Copyright 2016 Tecnativa S.L. - Pedro M. Baeza
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, exceptions, fields, models, _
+
+
+class ResPartnerIndustry(models.Model):
+    _inherit = 'res.partner.industry'
+    _order = "parent_left"
+    _parent_order = "name"
+    _parent_store = True
+
+    name = fields.Char(required=True)
+    parent_id = fields.Many2one(comodel_name='res.partner.industry',
+                                ondelete='restrict')
+    child_ids = fields.One2many(comodel_name='res.partner.industry',
+                                inverse_name='parent_id',
+                                string="Children")
+    parent_left = fields.Integer('Parent Left', index=True)
+    parent_right = fields.Integer('Parent Right', index=True)
+
+    @api.multi
+    def name_get(self):
+        def get_names(cat):
+            """ Return the list [cat.name, cat.parent_id.name, ...] """
+            res = []
+            while cat:
+                res.append(cat.name)
+                cat = cat.parent_id
+            return res
+
+        return [(cat.id, " / ".join(reversed(get_names(cat)))) for cat in self]
```

Comment:
So, you do a loop of a reversed loop of a loop... This sounds like a good candidate for a performance bottleneck. :thinking: 

I suggest:

```python
display_name = fields.Char(compute="_compute_display_name", store=True, index=True)

@api.depends("name", "parent_id")
def _compute_display_name(self):
    # Make your loops here; then:
    self.mapped("child_ids")._compute_display_name()
```

Storing the field will make the hyperloop only happen once on changes, and will allow better searching.

Reply:
Creating new industries is not something you do usually, just one day in your life. So, I think it's not justified the full performance approach by storing a <code>display_name</code> field. Also, maybe it would need a pre-migration script or a <code>pre_init_hook</code> to fill this field for those that have already using this module without this new field.

At least, I improved the performance by deleting the reversed loop.